control-plane/connect-inject/webhook/consul_dataplane_sidecar.go

```diff
@@ -314,7 +314,11 @@ func (w *MeshWebhook) getContainerSidecarArgs(namespace corev1.Namespace, mpi mu
 
 	// If Consul DNS is enabled, we want to configure consul-dataplane to be the DNS proxy
 	// for Consul DNS in the pod.
-	if w.EnableConsulDNS {
+	dnsEnabled, err := consulDNSEnabled(namespace, pod, w.EnableConsulDNS, w.EnableTransparentProxy)
```

Comment:
Bugfix is here. Use `consulDNSEnabled` instead of just checking global DNS setting. @ishustava can you confirm this change is okay. Previously we only looked at the global setting but I see there is also pod and namespace settings so I made the change to look at those too (in addition to checking if tproxy is enabled)

Reply:
Perfect, this is exactly what I was hoping we'd do.